src/commands/cmd_stream.cc

```diff
@@ -33,6 +33,22 @@
 #include "types/redis_stream.h"
 
 namespace redis {
+namespace {
+// for XRead and XReadGroup stream range parse.
+CommandKeyRange ParseStreamReadRange(const std::vector<std::string> &args) {
+  // assert here we must have a stream in args since it has been parsed.
+  auto stream_keyword_iter =
+      std::find_if(args.cbegin(), args.cend(), [](const std::string &arg) { return util::EqualICase(arg, "streams"); });
```

Comment:
For XREADGROUP, I think we need to find from `begin + 4`, to avoid a group or consumer named `streams`.

refer to https://redis.io/docs/latest/commands/xreadgroup/ .

Reply:
Hi @PragmaTwice ,

Thanks very much for your review! 😊

I have updated the `ParseStreamReadRange` with a begin offset.

Best Regards,
Edward